proxy/src/serverless/http_conn_pool.rs

```diff
@@ -210,10 +214,15 @@ impl<C: ClientInnerExt + Clone> GlobalConnPool<C> {
         self: &Arc<Self>,
         ctx: &RequestMonitoring,
         conn_info: &ConnInfo,
-    ) -> Option<Client<C>> {
-        let endpoint = conn_info.endpoint_cache_key()?;
+    ) -> Result<Option<Client<C>>, HttpConnError> {
+        // use identity to pass Result and unify the GlobalConnPool in future
+        let Some(endpoint) = conn_info.endpoint_cache_key() else {
+            return identity(Ok(None));
```

Comment:
Rather than using `identity`, can we put an `expect` attribute on this function to silence the unnecessary Result warning? You can provide a `reason = "..."` too so we can replace the associated comment